test/integration/smoke/test_host_maintenance.py

```diff
@@ -254,25 +301,101 @@ def test_02_cancel_host_maintenace_with_migration_jobs(self):
                 self.logger.debug("Creating vms = {}".format(no_vm_req))
                 self.vmlist = self.createVMs(listHost[0].id, no_vm_req)
         
-        vm_migrating=False
+        migrations_finished = True
         
         try:
-           
-           vm_migrating = self.hostPrepareAndCancelMaintenance(listHost[0].id, listHost[1].id, self.checkVmMigratingOnHost)
-           
-           vm_migrating = self.hostPrepareAndCancelMaintenance(listHost[1].id, listHost[0].id, self.checkVmMigratingOnHost)
+            migrations_finished = self.hostPrepareAndCancelMaintenance(listHost[0].id, listHost[1].id)
+
+            if migrations_finished:
+                migrations_finished = self.hostPrepareAndCancelMaintenance(listHost[1].id, listHost[0].id)
            
         except Exception as e:
             self.logger.debug("Exception {}".format(e))
             self.fail("Cancel host maintenance failed {}".format(e[0]))
-        
 
-        if (vm_migrating == False):
-            raise unittest.SkipTest("No VM is migrating and the test will not be able to check the conditions the test is intended for");
-                
-            
+
+        if (migrations_finished == False):
```

Comment:
Same here, no assertions in the tests. We need to set an expected outcome of the tests and make and assertion that migrations will be complete and vms on new host for example, not clear what should the outcome of this test be.

Reply:
self.hostPrepareAndCancelMaintenance internally does the checking for us. This structure was already laid down.